.github/workflows/pre-commit.yaml

```diff
@@ -0,0 +1,30 @@
+name: pre-commit checks
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+      - ready_for_review
+  push:
+    branches:
+      - main
+env:
+  GO_VERSION: '^1.20'
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3.3.1
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - run: go version
+      - run: go install golang.org/x/tools/cmd/goimports@latest
+      - run: go install golang.org/x/lint/golint@latest
+      - run: curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -  -b $(go env GOPATH)/bin v1.51.2
+      - run: echo "PATH=$PATH:/home/runner/go/bin" >> $GITHUB_ENV
```

Comment:
Why do we need all these Go dependencies and golangci-lint? 

Fwiw, golangci-lint has [its own action](https://github.com/strangelove-ventures/cosmos-operator/blob/52504cc80c4a263b25aede7c5149f7db24fdef78/.github/workflows/go.yaml#L18) that works well.

Reply:
i thought this was needed for running the yaml lint, i can retest with removing these lines